### PR TITLE
chore: improvements on mm_utils

### PR DESCRIPTION
**Pull request by mickqian**
## Modifications

1. replace `expand_as` + `masked_scatter`(out-of-place) with  `torch.where` + in-place replacement, which sees 1-time speedup in regular image request scenarios:

```
VOCAB_SIZE = 32000
HIDDEN_SIZE = 3584
TOTAL_TOKENS = 2962  # Represents a flattened batch of sequences
NUM_MM_TOKENS = 512  # Number of multimodal tokens to insert

--- Test Configuration ---
Device: CUDA
Repetitions: 1000 (Warm-up: 10)
Tensor Shape: (2962, 3584) with 512 scattered elements.
----------------------------
Method: masked_scatter (out-of-place)
Total time for 1000 reps: 181.40 ms
Average time per call: 0.1814 ms

Method: Advanced Indexing (in-place)
Total time for 1000 reps: 75.95 ms
Average time per call: 0.0760 ms
```
2. some minor device-sync removements

## Checklist

- [ ] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://slack.sglang.ai to discuss your PR.

### CODE REVIEW

**Review by JustinTong0323 — APPROVED** [review]
LGTM

**Review by JustinTong0323 — APPROVED** [review]
